src/components/AppBar/ScheduleLoginManager.js

```diff
@@ -1,31 +1,122 @@
 
     useEffect(() => {
         const setupUser = async () => {
+            // See if there is a google user
             const user = await checkUser();
             setUser(user);
-            if (!user || !(await loadGoogleUser(user))) {
-                if (typeof Storage !== 'undefined') {
-                    const savedUserID = window.localStorage.getItem('userID');
-                    if (savedUserID != null) {
-                        loadSchedule(savedUserID, true);
-                    }
+
+            // Load the Google user's schedule
+            // didLoadGoogleUser is a boolean representing successful load
+            const didLoadGoogleUser = user && (await loadGoogleUser(user));
+
+            if (typeof Storage !== 'undefined') {
+                // HANDLING LEGACY SCHEDULES
+                // If there is no google user but a legacy schedule exists,
+                // load it that schedule
+                const legacyUserId = window.localStorage.getItem('userID');
+                if (!didLoadGoogleUser && legacyUserId != null) {
+                    loadSchedule(legacyUserId, true);
+                }
+
+                // HANDLING TEMP USER DATA
+                // If there is tempUserData, those schedules should be appended onto the userData
+                // After we've successfully loaded that temporary schedule, we can remove it from local storage
+                const tempUserData = window.localStorage.getItem('tempUserData');
+                if (tempUserData != null) {
+                    const userData = compileUserData();
+                    const oldData = JSON.parse(tempUserData);
+
+                    oldData.addedCourses = oldData.addedCourses.map((course) => {
+                        // Course schedule indices must be offset to account for existing schedules
+                        // Add the current number of schedules onto the index
+                        return {
+                            ...course,
+                            scheduleIndices: course.scheduleIndices.map(
+                                (index) => (index += userData.scheduleNames.length)
+                            ),
+                        };
+                    });
+                    oldData.customEvents = oldData.customEvents.map((event) => {
+                        // Custom event indices must be offset to account for existing schedules
+                        // Add the current number of schedules onto the index
+                        return {
+                            ...event,
+                            scheduleIndices: event.scheduleIndices.map(
+                                (index) => (index += userData.scheduleNames.length)
+                            ),
+                        };
+                    });
+
+                    // Add the cached data onto userData
+                    userData.addedCourses = userData.addedCourses.concat(oldData.addedCourses);
```

Comment:
I don't think this works if you already have a course in your Google schedule and you try to add the same course from the old schedule. Instead we would have to add the schedule indices if the same course is in both the old and new schedules.

Reply:
This works because of the lines above 50-70. It creates a new schedule rather than overwrite the schedules from the google login.

Reply:
Oh I see the problem